config.go

```diff
@@ -101,6 +107,10 @@ type Config struct {
 	// LocalAddr is the local address to use when dialing an nsqd.
 	// If empty, a local address is automatically chosen.
 	LocalAddr net.Addr `opt:"local_addr"`
+	// Dialer affect connection when dialing an nsqd. Overwrite this to connect over proxy.
+	//
+	// Conflict with options LocalAddr and DialTimeout.
+	Dialer Dialer `opt:"dialer"`
```

Comment:
We need to decide how to handle configuration specified for this `opt`, e.g. for command line utilities in `nsqd` that use `--consumer-opt`.

See: https://github.com/nsqio/nsq/blob/master/apps/nsq_to_file/nsq_to_file.go#L131-L135

And an example of how we handle `tls_config`: https://github.com/nsqio/go-nsq/blob/master/config.go#L388-L475